src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -228,6 +231,31 @@ public Boolean getUserProperty(String userId, String recipeId, String mapName, T
     return getRecipeMetadataQuery(recipesQuery, sortingMethod);
   }
 
+  // TODO: Currently, sorting method is unused. Used TOP in recipesMatchingAnyTags() for consistent
+  // order.
+  public List<RecipeMetadata> getRecipesMatchingFollowedTags(
+      String userId, SortingMethod sortingMethod, int page) {
+    List<RecipeMetadata> followedTagsRecipes = recipesMatchingAnyTags(followedTagIds(userId));
+    // Return the appropriate page of recipes manually.
+    try {
+      return followedTagsRecipes.subList(
+          (page * RECIPES_PER_PAGE), ((page + 1) * RECIPES_PER_PAGE));
+    } catch (IndexOutOfBoundsException e) {
+      if (page == 0) {
+        return followedTagsRecipes;
+      } else {
+        return null;
+      }
+    }
+  }
+
+  public List<RecipeMetadata> getRecipesMatchingFollowedTags(
+      String userId, SortingMethod sortingMethod) {
+    List<String> followedTagIds = followedTagIds(userId);
+    // Query for recipes matching the followed tag Ids.
+    return recipesMatchingAnyTags(followedTagIds);
+  }
```

Comment:
If you want to do sorting, you could add some code like this (does sort on our end without firestore's help):
```
switch (sortingMethod) {
      case TOP:
        System.out.println("Sorting by: TOP");
        Collections.sort(
            results, Collections.reverseOrder(Comparator.comparingLong(RecipeMetadata::getVotes)));
        break;
      case NEW:
        System.out.println("Sorting by: NEW");
        Collections.sort(results,
            Collections.reverseOrder(Comparator.comparingLong(RecipeMetadata::getTimestamp)));
        break;
    }
```

Reply:
Here, getVotes and getTimestamp are simple getters I added on my branch